### ISSUE TITLE
OSError: [WinError 1] When running `meson <builddir>`

### ISSUE THREAD

**Issue by ClayCore** [mesonbuild/meson#10681]
**Describe the bug**
After running `meson <builddir>`  and once meson finds ninja, I encounter a critical `OSError: [WinError 1] Invalid function` with a corrupted path to some mingw64 library.

Once the stacktrace and error message is printed out I am presented with the following message:

```
ERROR: Unhandled python exception
    This is a Meson bug and should be reported!
```

I've attached the stacktrace to this issue and to the repository.

I believe that perhaps ninja appears to fail to resolve where my compiler lives, which wasn't an issue before.
What I find particularly interesting is how it tries to resolve `D:\` when none of my drives are assigned to this letter.

**To Reproduce**
[Repository with the minimal example](https://github.com/ClayCore/meson-issue)

**Expected behavior**
Proper configuration of the project.

**System Parameters:**
**build:** native package from `pip install meson`
**operating system:** Windows 10 LTSC (1809)
**python version:** Python 3.10
**meson version:** 0.63.0
**ninja version:** 1.10.2.git.kitware.jobserver-1
**compiler:** gcc.exe by MSYS2 12.1.0

[stacktrace.txt](https://github.com/mesonbuild/meson/files/9320841/stacktrace.txt)

**Comment by eli-schwartz** [comment]
I feel like I've seen another ticket about this, but I can't find it. Something to do with the compiler embedding some runtime paths based on the build-time environment.

Anyway, perhaps @lazka has an idea of what's happening here.

**Comment by lazka** [comment]
Yeah, there is ancient gcc bug where it outputs the build path (because it needs it during building itself afair...), and meson tries to filter them out, and then fails there.

@ClayCore is there anything special about your "D:" drive?

**Comment by ClayCore** [comment]
Nope. It's an external CD drive and thats really it. My msys2 toolchain is installed in S:, which is a regular hard drive

**Comment by lazka** [comment]
I would call a "CD drive" special :) That's likely the trigger then.

**Comment by ClayCore** [comment]
After disconnecting the drive meson manages to configure the project fine! Still seems pretty weird to me, since I had that drive plugged in for as long as I can remember, and when I first started using meson it had no troubles configuring and compiling the project
